### DISCUSSION TITLE
为什么Table组件需要采用rc-table了？

### DISCUSSION THREAD

**Discussion by JackDan9** [ant-design/ant-design#32851] [Q&A] [answered]
我在阅读[table](https://github.com/ant-design/ant-design/blob/master/components/table/Table.tsx)源码的过程中发现，引入了[rc-table](https://github.com/react-component/table/blob/master/src/Table.tsx)代码，我想为什么要引入rc-table的组件库作为ant-design的table依赖了？其实可以自己写一个的了，我自己的见解，因为rc-table的问题也会引入到ant-design的table中，有哪位大佬能帮我解答一下疑惑吗？非常感谢。

**Comment by yoyo837** [comment]
你仔细看package.json可以发现大部份功能都提取到rc-*的包里了，他们也是antd团队维护的，提取的原因主要是方便快速patch.

**Comment by yoyo837** [comment, marked as answer]
而且rc-*基本不包含样式，可以复用到不同的风格

**Reply by JackDan9** [reply]
感谢，但是rc-*本身的问题也会服用到ant-design中，我简单举一个例子哈，rc-table的scroll属性设置为true或者max-content之后，width的属性将不会生效了，但是看ant-design的[table](https://ant.design/components/table-cn/#scroll)文档解释并没有备注这一项，我认为这其实是一个文档上的Bug了，或者是否可以认领去修复一下，rc-table不支持，但是经过ant-design的封装它支持了。
